src/app/clusters/door-lock-server/door-lock-server.cpp

```diff
@@ -3812,6 +3871,108 @@ bool emberAfDoorLockClusterClearHolidayScheduleCallback(
+{
+    DoorLockServer::Instance().setAliroReaderConfigCommandHandler(commandObj, commandPath, commandData.signingKey,
+                                                                  commandData.verificationKey, commandData.groupIdentifier,
+                                                                  commandData.groupResolvingKey);
+    return true;
+}
+
+bool emberAfDoorLockClusterClearAliroReaderConfigCallback(CommandHandler * commandObj, const ConcreteCommandPath & commandPath,
+                                                          const Commands::ClearAliroReaderConfig::DecodableType & commandData)
+{
+    DoorLockServer::Instance().clearAliroReaderConfigCommandHandler(commandObj, commandPath);
+    return true;
+}
+
+void DoorLockServer::setAliroReaderConfigCommandHandler(CommandHandler * commandObj, const ConcreteCommandPath & commandPath,
+                                                        const ByteSpan & signingKey, const ByteSpan & verificationKey,
+                                                        const ByteSpan & groupIdentifier,
+                                                        const Optional<ByteSpan> & groupResolvingKey)
+{
+    EndpointId endpointID = commandPath.mEndpointId;
+    ChipLogProgress(Zcl, "[SetAliroReaderConfig] Incoming command [endpointId=%d]", endpointID);
+
+    // If Aliro Provisioning feature is not supported, return UNSUPPORTED_COMMAND.
+    if (!SupportsAliroProvisioning(endpointID))
+    {
+        ChipLogProgress(Zcl, "[SetAliroReaderConfig] Aliro Provisioning is not supported [endpointId=%d]", endpointID);
+        commandObj->AddStatus(commandPath, Status::UnsupportedCommand);
+        return;
+    }
+
+    Delegate * delegate = GetDelegate(endpointID);
+    VerifyOrReturn(delegate != nullptr, ChipLogError(Zcl, "Delegate is null"));
+
+    // If Aliro BLE UWB feature is supported and groupResolvingKey is not provided in the command, return INVALID_COMMAND.
+    if (SupportsAliroBLEUWB(endpointID) && !groupResolvingKey.HasValue())
+    {
+        ChipLogProgress(Zcl, "[SetAliroReaderConfig] Aliro BLE UWB supported but Group Resolving Key is not provided");
+        commandObj->AddStatus(commandPath, Status::InvalidCommand);
+        return;
+    }
+
+    // Check if the size of the signingKey, verificationKey, groupIdentifier, groupResolvingKey parameters conform to the spec.
+    // Return INVALID_COMMAND if not.
+    if (signingKey.size() != kAliroSigningKeySize || verificationKey.size() != kAliroReaderVerificationKeySize ||
+        groupIdentifier.size() != kAliroReaderGroupIdentifierSize ||
+        (groupResolvingKey.HasValue() && groupResolvingKey.Value().size() != kAliroGroupResolvingKeySize))
+    {
+        ChipLogProgress(Zcl,
+                        "[SetAliroReaderConfig] One or more parameters in the command do not meet the size constraint as per spec");
+        commandObj->AddStatus(commandPath, Status::InvalidCommand);
```

Comment:
```suggestion
        commandObj->AddStatus(commandPath, Status::ConstraintError);
```
is what the spec says, no?